persistcache/persistcache.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2023 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package persistcache
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+type persistCache struct {
+	cache map[string]string
+	root  string
+}
+
+const FILE_MASK = 0755
+
+// Load values from cache or creates path if there's none
+func Load(path string) (*persistCache, error) {
+	pc := &persistCache{}
+	pc.root = path
+	pc.cache = make(map[string]string)
+
+	if _, err := os.Stat(pc.root); os.IsNotExist(err) {
+		os.MkdirAll(pc.root, FILE_MASK)
+		return pc, nil
+	}
+
+	err := filepath.WalkDir(pc.root, func(path string, di fs.DirEntry, err error) error {
+		// We skip all directories
+		if di.IsDir() {
+			return nil
+		}
+
+		val, err := os.ReadFile(path)
+		if err != nil {
+			return err
+		}
+
+		pc.cache[di.Name()] = string(val)
+
+		return nil
+	})
+	return pc, err
+}
+
+// Get value from cache
+func (pc *persistCache) Get(key string) (string, bool) {
+	val, ok := pc.cache[key]
+	return val, ok
+}
+
+// Create or update value in in-memory cache and filesystem
+func (pc *persistCache) Put(key string, val string) (string, error) {
+	pc.cache[key] = val
+
```

Comment:
Can you filter out '/' out of `key`? In case somebody comes with a key like `../../../../../../etc/passwd`.

Reply:
That's a good catch, will do

Reply:
but what if somebody will use /etc as root and try to overwrite passwd ?

Reply:
Well, but then the developer has chosen to use `/etc` ... which might be a valid choice f.e. on Windows systems ...

My biggest concern is that `key` is somehow dependent on a user's choice and it would allow them to overwrite arbitrary files.

Reply:
Btw. Same problem with `Get(key string)`

Reply:
Well, current implementation is using local cache to get values, so it won't be a problem since you can only get value from folder they specify or from values they Put. But we will have to check for valid path when we do lazy initialization

Reply:
ah yeah, you're right!